test/e2e/all_in_one_test.go

```diff
@@ -89,6 +97,76 @@ const (
 	dblessURL  = "https://raw.githubusercontent.com/Kong/kubernetes-ingress-controller/%v.%v.x/deploy/single/all-in-one-dbless.yaml"
 )
 
+func TestDeployAllInOneDBLESSGateway(t *testing.T) {
+	t.Log("configuring all-in-one-dbless.yaml manifest test")
+	t.Parallel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	t.Log("building test cluster and environment")
+	addons := []clusters.Addon{}
+	addons = append(addons, metallb.New())
+	if b, err := loadimage.NewBuilder().WithImage(imageLoad); err == nil {
+		addons = append(addons, b.Build())
+	}
+	builder := environments.NewBuilder().WithAddons(addons...)
+	if clusterVersionStr != "" {
+		clusterVersion, err := semver.ParseTolerant(clusterVersionStr)
+		require.NoError(t, err)
+		builder.WithKubernetesVersion(clusterVersion)
+	}
+	env, err := builder.Build(ctx)
+	require.NoError(t, err)
+	defer func() {
+		assert.NoError(t, env.Cleanup(ctx))
+	}()
+
+	require.NoError(t, clusters.KustomizeDeployForCluster(ctx, env.Cluster(), "github.com/kubernetes-sigs/gateway-api/config/crd?ref=v0.4.0"))
```

Comment:
I think we should provide a helpful log line here about what we're trying to do, it will make reading a failure to deploy the CRDs here easier to read:

```suggestion
	t.Log("deploying Gateway APIs CRDs from %s", gatewayCRDsURL)
	require.NoError(t, clusters.KustomizeDeployForCluster(ctx, env.Cluster(), gatewayCRDsURL))
```

Also I think we should place the URL in a const:

```
const gatewayCRDsURL = "github.com/kubernetes-sigs/gateway-api/config/crd"
```

And at least in the short term I think we should remove the explicit version `?ref=v0.4.0` from the URL: this will allow this nightly E2E test to be a smoke signal for us if upstream somehow breaks the CRDs and we would know within a day, and also avoid us having to manually update the version.